cli/utilities/utils.py

```diff
@@ -56,6 +59,70 @@ def config_dict_to_string(data):
     return rtn
 
 
+def get_nodes_by_ids(nodes, node_names):
+    """
+    Fetch nodes using provided substring of nodes
+    Returns node list which matched node ID(eg., 'node1')
+    else all nodes object list if node_names is empty.
+    Args:
+        nodes: Cli obj containing node details
+        node_names: node name list (eg., ['node1'])
+    Returns:
+        node_list: list of nodes
+    """
+
+    node_details = []
+    for name in node_names:
+        node = get_node_by_id(nodes, name)
+        if node:
+            node_details.append(node)
+    return node_details
+
+
+def get_node_by_id(nodes, node_name):
+    """
+    Fetch node using provided node substring::
+        As per the naming convention used at VM creation, where each node.shortname
+        is framed with hyphen("-") separated string as below, please refer
+        ceph.utils.create_ceph_nodes definition
+            "ceph-<name>-node1-<roles>"
+            name: RHOS-D username or provided --instances-name
+            roles: roles attached to node in inventory file
+        In this method we use hyphen("-") appended to node_name string to try fetch exact node.
+        for example,
+            "node1" ----> "node1-"
+        Note: But however if node_name doesn't follow naming convention as mentioned in
+        inventory, the first searched node will be returned.
+        for example,
+            "node" ----> it might return any node which matched first, like node11.
+        return None, If this cluster has no nodes with this substring.
+    Args:
+        nodes: Cli obj containing node details
+        node_name: node1        # try to use node<Id>
+    Returns:
+        node instance (CephVMNode)
+    """
+    for node in nodes:
+        searches = re.findall(rf"{node_name}?\d*", node.hostname, re.IGNORECASE)
+        for ele in searches:
+            if ele == node_name:
+                return node
+    return None
+
+
+def build_cmd_from_args(**kw):
```

Comment:
This can be improved for parameters using `--<parameter-name>` and `--<parameter-name>=`

Reply:
Added